spec/support/shared_examples/container.rb

```diff
@@ -185,6 +185,26 @@
           expect(container[:item].call).to eq('item')
         end
       end
+
+      context 'with option lazy_singleton: true' do
+        it 'registers and resolves a proc' do
+          container.register(:item, proc { 'item' }, lazy_singleton: true)
+
+          expect(container.keys).to eq(['item'])
+          expect(container.key?(:item)).to be true
+          expect(container.resolve(:item)).to eq('item')
+          expect(container[:item]).to eq('item')
+        end
```

Comment:
Yes indeed. I was using this spec only to test that the normal behavior was still supported, and was using the one after to test that the initialization was made only once, but I agree with your changes, as they tighten the spec.